File: app/src/main/java/com/codeme/pdm/MainActivity.java
package com.codeme.pdm;

import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import java.io.File;
import java.io.FileOutputStream;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    List<Fragment> fragmentList = null;
    ArrayList<String> titleList = null;
    SectionsPagerAdapter spa;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        openDatabase();
        App.db = SQLiteDatabase.openOrCreateDatabase(App.databaseFilename,null);

        fragmentList = new ArrayList<>();
        fragmentList.add(new fragment1());
        fragmentList.add(new fragment2());

        titleList = new ArrayList<>();
        titleList.add("供应商");
        titleList.add("产品清单");

        spa = new SectionsPagerAdapter(getSupportFragmentManager(),fragmentList, titleList);
        ViewPager mViewPager = (ViewPager) findViewById(R.id.container);
        mViewPager.setAdapter(spa);

        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabs);
        tabLayout.setupWithViewPager(mViewPager);

    }
    private void openDatabase() {
//        app.DATABASE_PATH=MainActivity.this.getFilesDir().toString();
        try {
            File dir = new File(App.DATABASE_PATH);
            if (!dir.exists())
                dir.mkdir();
            if (!(new File(App.databaseFilename)).exists()) {
                // 获得封装dictionary.db文件的InputStream对象
                InputStream is = getResources().openRawResource(R.raw.pdm);
                FileOutputStream fos = new FileOutputStream(App.databaseFilename);
                byte[] buffer = new byte[7168];
                int count;
                // 开始复制dictionary.db文件
                while ((count = is.read(buffer)) > 0) {
                    fos.write(buffer, 0, count);
                }
                (Toast.makeText(this, "数据库创建成功",Toast.LENGTH_LONG)).show();
                fos.close();
                is.close();
            }
        } catch (Exception e) {
            (Toast.makeText(this, "数据库创建错误：" + e.getMessage(),
                    Toast.LENGTH_LONG)).show();
        }
    }
}

File: app/src/main/java/com/codeme/pdm/MainActivity.kt
package com.codeme.pdm

import android.database.sqlite.SQLiteDatabase
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.io.File
import java.io.FileOutputStream
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (openDatabase())
            App.db = SQLiteDatabase.openOrCreateDatabase(App.databaseFilename, null)

        val fragmentList = ArrayList<Fragment>()
        fragmentList.add(Fragment1())
        fragmentList.add(Fragment2())
        fragmentList.add(Fragment3())

        val titleList = ArrayList<String>()
        titleList.add("供应商")
        titleList.add("认可报告")
        titleList.add("产品清单")

        App.spa = SectionsPagerAdapter(supportFragmentManager, fragmentList, titleList)
        container.adapter = App.spa
        tabs.setupWithViewPager(container)
    }

    private fun openDatabase(): Boolean {
        App.DATABASE_PATH = "${this.filesDir}"
        App.databaseFilename = App.DATABASE_PATH + "/pdm.db"
        try {
            val dir = File(App.DATABASE_PATH)
            if (!dir.exists())
                dir.mkdir()
            if (!File(App.databaseFilename).exists()) {
                // 获得封装dictionary.db文件的InputStream对象
                val `is` = resources.openRawResource(R.raw.pdm)
                val fos = FileOutputStream(App.databaseFilename)
                val buffer = ByteArray(7168)
                var count: Int = `is`.read(buffer)
                // 开始复制dictionary.db文件
                while (count > 0) {
                    fos.write(buffer, 0, count)
                    count = `is`.read(buffer)
                }
                Toast.makeText(this, "数据库创建成功", Toast.LENGTH_LONG).show()
                fos.close()
                `is`.close()
            }
        } catch (e: Exception) {
            Toast.makeText(this, "数据库创建错误：" + e.message,
                    Toast.LENGTH_LONG).show()
            return false
        }
        return true
    }
}
